### PR TITLE
feat: implement quantile in more backends

### PR DESCRIPTION
**Pull request by cpcloud**
This PR implements the `Quantile` and `MultiQuantile` (where supported by the
backend) for the ClickHouse, DuckDB and PostgreSQL backends.

### CODE REVIEW

**Comment by cpcloud** [comment]
I'm going to remove the bits that make this operation filterable into a separate PR since that's a breaking change.

File: ibis/expr/analysis.py [hunk 1]
```diff
@@ -201,7 +201,10 @@ def substitute(fn, expr):
     """
     Substitute expressions with other expressions.
     """
-    node = expr.op()
+    try:
```
Comment (kszucs) [inline ibis/expr/analysis.py:204]: Is this required?

**Review by kszucs — APPROVED** [review]
Just a nit otherwise LGTM

File: ibis/expr/analysis.py (hunk 1 above)
Comment (cpcloud) [reply ibis/expr/analysis.py:204]: Only if we merge this before #4015. I'll remove it so this PR fails until we merge that one.

**Pull request closed without merging**